### ISSUE TITLE
Method signature mismatch with MultiLanguageLG using Adaptive Dialogs while generating LG

### ISSUE THREAD

**Issue by dschenzer** [microsoft/botbuilder-js#3869]
## Versions
What package version of the SDK are you using.
-> 4.13.5 & 4.13.5-preview for "botbuilder-dialogs-adaptive", "botbuilder-dialogs-declarative"

What nodejs version are you using
v14.15.4 & NPM 7.5.3

What browser version are you using
MS Edge 91

What os are you using
Windows 10 & macOS

## Describe the bug
Given we use Adaptive Dialogs in pure JavaScript and the **MultiLanguageLG**-class to declare a multi-language aware dialog:

The generate-function with the **MultiLanguageLG** has a different method signature then **TemplateEngineGenerator** (_which works with one LG file_). 
Our requirement is to use a multi-language LG files and serve it properly to Adaptive Dialogs.
![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/5458637/125766246-ae6816fb-a178-4922-b326-ecb2ccf4e027.png)

## To Reproduce
1. Create a **MultiLanguageLG** with multiple languages
2. Assign it to a AdaptiveDialog.generator
3. When the AdaptiveDialog is generating LG, it should throw

## Expected behavior
When MultiLanguageLG is assigned to an AdaptiveDialog (_either dynamically loaded from dialog-file or constructed in code manually_) the multi-language aware generation doesn't throw. 
When I swap the parameters (template and dialog context) of the generate-function it seems to be Ok. 

## Screenshots
![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/5458637/125768846-65474432-5321-4649-806e-7d7b96796d6f.png)

## Additional context
When an AdaptiveDialog is invoking the language generation is seem to be not aware of the MultiLanguageLG.generate-function that seem to have a different order of parameters. When using TemplateEngineGenerator, the language generation seems to work. 

We are not using the Bot Composer yet for this part. All dialogs are in JavaScript or .dialog-files (and loaded on startup). 

This is an blocking issue for our customer.

**Comment by Danieladu** [comment]
Sorry for the confusion.
 
`MultiLanguageLG` and `MultiLanguageGenerator` not directly related.

The `MultiLanguageLG` is an independent class that owned by LG library.  It accepts the "locale to Templates mapping" or "locale to file path mapping", and a default locale.  User could use the "generate" (provide template name, memory data, and the current locale)  to get the multi-lang result. Here is the test/usage of  `MultiLanguageLG` : https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder-lg/tests/multilanguageLG.test.js

But for `MultiLanguageGenerator`, it accepts a dictionary that contains locale to `LanguageGenerator` mapping. The `LanguageGenerator` could be `TemplateEngineLanguageGenerator`.
Here are some samples/tests for `MultiLanguageGenerator`:https://github.com/microsoft/botbuilder-js/blob/285f3e9457fa9e76de2b891948e7f494aee3d101/libraries/botbuilder-dialogs-adaptive/tests/LGGenerator.test.js#L122

In your scenario, define an MultiLanguageGenerator is the right way, and assign each locale generator to `languageGenerators` could make it real-multi-lang. For example:
```
const generator = new MultiLanguageGenerator();
lg.languageGenerators.set('de', new TemplateEngineLanguageGenerator());
lg.languageGenerators.set('en-us', new TemplateEngineLanguageGenerator(Templates.parseFile('xxx/1.lg')));
...
const result = generator.generate(dialogContext, templatebody, data);
```

For the error: `dialogContext.getLocale is not a function`, make sure bump the "botbuilder-dialog" library to align with `botbuilder-dialogs-adaptive` library.

Thanks.

**Comment by taicchoumsft** [comment]
@dschenzer Does @Danieladu 's workaround work for you?

**Comment by dschenzer** [comment]
@Danieladu Sorry, for late response. I was on vacations and just back. Currently I'm stuck with another issue that suddenly arise in the bot framework library so that I'm blocked to progress on this and test it.

**Comment by dschenzer** [comment]
I have started working on this issue again. I have updated our LG components to use `MultiLanguageGenerator` while traversing all the files in our directory. 
Now I wanted to generate LG just for a greeting and it seems like we are missing some services in the `DialogContext`. 

Do we need to do some setup in the BotAdapter? 

![image](https://user-images.githubusercontent.com/5458637/128694373-6ee18d8b-3628-41a0-a655-13e29e8c5c48.png)

Package.json:
![image](https://user-images.githubusercontent.com/5458637/128694549-3c388870-4ec1-40a7-b424-a3b013573f96.png)
